api_v3/services/FlavorAssetService.php

```diff
@@ -923,4 +923,156 @@ public function deleteLocalContentAction($assetId)
 			$fileSync->save();
 		}
 	}
-}
\ No newline at end of file
+
+	/**
+	 * serve cmd line to transcode the ad
+	 *
+	 * @action serveAdStitchCmd
+	 * @param string $assetId
+	 * @param string $mediaInfoJson
+	 *
+	 * @throws KalturaAPIException
+	 */
+	public function serveAdStitchCmdAction($assetId , $mediaInfoJson)
+	{
+		$asset = assetPeer::retrieveById($assetId);
+		if(is_null($asset))
+			throw new KalturaAPIException(KalturaErrors::ASSET_ID_NOT_FOUND, $assetId);
+
+		$flavorParamsId = $asset->getFlavorParamsId();
+
+		$flavorParamsDb = assetParamsPeer::retrieveByPK($flavorParamsId);
+
+		if (!$flavorParamsDb)
+			throw new KalturaAPIException(KalturaErrors::FLAVOR_PARAMS_ID_NOT_FOUND, $flavorParamsId);
+
+		$flavorParamsOutputDb = assetParamsOutputPeer::retrieveByAssetId($assetId);
+
+		if (!$flavorParamsOutputDb)
+			throw new KalturaAPIException(KalturaErrors::FLAVOR_PARAMS_OUTPUT_ID_NOT_FOUND, $assetId);
+
+		$cmdLine = self::generateAdStitchingCmdline($mediaInfoJson, $flavorParamsDb, $flavorParamsOutputDb);
+
+		return $cmdLine;
+	}
+	
+	private static function generateAdStitchingCmdline($ffprobeJson, $flavorParams, $flavorParamsOutput)
+	{
+		$parser = new KFFMpegMediaParserAdStitchHelper($ffprobeJson);
+		$srcMedInf = $parser->getMediaInfo();
+		$srcMedSet = KFFMpegMediaParserAdStitchHelper::mediaInfoToKDL($srcMedInf);
+
+/**
+ * To match from flavor params output-
+ * - frame size
+ * - fps
+ * - gop
+ * - audio params
+ * Other matching
+ * - AvoidVideoShrinkFramesizeToSource = on
+ * - two pass - off
+ * - encypted - off
+ * - letter boxing
+ * - other extra params (from live sticthing)
+ */
+		$kdlFlavor = KDLWrap::ConvertFlavorCdl2Kdl($flavorParams);
+		$kdlFlavor->_isTwoPass = false;
+		$kdlFlavor->_isEncrypted = false;
+		$kdlFlavor->_video->_arProcessingMode = 2; // letter boxing
+		$kdlFlavor->_video->_isShrinkFramesizeToSource = false;
+		if($flavorParamsOutput->getWidth()){
+			$kdlFlavor->_video->_width = $flavorParamsOutput->getWidth();
+		}
+		if($flavorParamsOutput->getHeight()){
+			$kdlFlavor->_video->_height = $flavorParamsOutput->getHeight();
+		}
+		if($flavorParamsOutput->getGopSize()){
+			$kdlFlavor->_video->_gop = $flavorParamsOutput->getGopSize();
+		}
+		if($flavorParamsOutput->getFrameRate()){
+			$kdlFlavor->_video->_frameRate = $flavorParamsOutput->getFrameRate();
+		}
+		if($flavorParamsOutput->getAudioBitrate()){
+			$kdlFlavor->_audio->_bitRate = $flavorParamsOutput->getAudioBitrate();
+		}
+		if($flavorParamsOutput->getAudioSampleRate()){
+			$kdlFlavor->_audio->_sampleRate = $flavorParamsOutput->getAudioSampleRate();
+		}
+		$target = $kdlFlavor->GenerateTarget($srcMedSet);
+		$cmdLine = $target->_transcoders[0]->_cmd;
+		return $cmdLine;
+	}
+}
+
+class KFFMpegMediaParserAdStitchHelper extends KFFMpegMediaParser{
```

Comment:
Also, this class belong to Business Convert Decision Layer: alpha/apps/kaltura/lib/batch2/bcdl